imap_processing/ialirt/l0/decom_ialirt.py

```diff
@@ -0,0 +1,142 @@
+import logging
+from enum import Enum
+
+import bitstring
+import xarray as xr
+from space_packet_parser import parser, xtcedef
+
+logging.basicConfig(level=logging.ERROR)
+logger = logging.getLogger(__name__)
+
+
+class PacketLength(Enum):
+    """Class that represents properties of the IALiRT packet."""
+
+    EXPECTED_LENGTH = 1464
+
+def decom_packets(packet_file, xtce_packet_definition):
+    """
+    Unpack data packet. In this function, we unpack and return data.
+
+    Parameters
+    ----------
+    packet_file : str
+        Path to data packet path with filename
+    xtce_packet_definition : str
+        Path to XTCE file with filename
+
+    Returns
+    -------
+    packets : list
+        List of all the unpacked data
+    """
+    packet_definition = xtcedef.XtcePacketDefinition(xtce_packet_definition)
+    packet_parser = parser.PacketParser(packet_definition)
+
+    packets = []
+
+    with open(packet_file) as file:
+        for line_number, line in enumerate(file, 1):
+            if not line.startswith("#"):
+                # Split the line by semicolons
+                # Discard the first value since it is only a counter
+                hex_values = line.strip().split(";")[1::]
+
+                binary_values = ""
+                for h in hex_values:
+                    # Convert hex to integer
+                    # 16 is the base of hexadecimal
+                    int_value = int(h, 16)
+
+                    # Convert integer to binary and remove the '0b' prefix
+                    bin_value = bin(int_value)[2:]
+
+                    # Make sure each binary string is 8 bits long
+                    bin_value_padded = bin_value.zfill(8)
+
+                    # Append the padded binary string to the final string
+                    binary_values += bin_value_padded
+
+                # Check the length of binary_values
+                if len(binary_values) != PacketLength.EXPECTED_LENGTH.value:
```

Comment:
Could add a few bytes to one of your test rows and create a new temporary file that you can then hit this code path with I think.